### DISCUSSION TITLE
Enable podman-auto-update.timer in Wiki?

### DISCUSSION THREAD

**Discussion by Klaaktu** [dani-garcia/vaultwarden#4362] [Ideas]
The wiki says:
> Additional benefit is that this method will keep the container updated.

However, from [podman-systemd.unit](https://docs.podman.io/en/latest/markdown/podman-systemd.unit.5.html#autoupdate), it seems that `AutoUpdate=registry` + `Image=...` only means that you can do `podman auto-update`. To auto update as in unattended, [podman-auto-update.timer](https://docs.podman.io/en/latest/markdown/podman-auto-update.1.html#systemd-unit-and-timer) needs to be enabled.

So the wiki could suggest either `sudo podman auto-update` once in a while or `sudo systemctl enable podman-auto-update.timer`.

p.s. I remember somewhere said that daemon-reload/reboot would trigger image update, but it doesn't seem to be true for me.

**Comment by Arcitec** [comment]
You are correct.

- Enabling `AutoUpdate=` does nothing by itself. It's just a marker to let Podman know that you want to include that container in the auto-update processing.
- `podman auto-update` is the actual command that checks and installs updates. You can run it manually anytime if you want.
- `podman-auto-update.service` is a "oneshot service" which just runs `podman auto-update` followed by `podman image prune -f` and then immediately exits itself. Enabling this "service" makes no sense and you should never do it!
- `podman-auto-update.timer` is an actual timer which triggers `podman-auto-update.service` every day. That is the correct thing to enable.
- To enable it for root: `sudo systemctl enable --now podman-auto-update.timer`
- Or if you run the container as a rootless user, enable it for the user with: `systemctl --user enable --now podman-auto-update.timer`
- The default timer behavior is to run at a slightly randomized time (within 15 minutes) after midnight every day. If you miss the timer date, Systemd timers will instead run the next time your computer boots/wakes up, so you never have to worry about "missing the timer".
- You can also edit the timer interval with the following commands:
- Editing timer for root: `sudo systemctl edit podman-auto-update.timer`
- Editing timer for user: `systemctl --user edit podman-auto-update.timer`
- Editing timers is complicated, but there's a helpful [article](https://fedoramagazine.org/systemd-timers-for-scheduling-tasks/) and [systemd.timer man page](https://www.freedesktop.org/software/systemd/man/latest/systemd.timer.html).
- If you want to see the root logs for the last time the auto-update service ran (doesn't include manual runs): `journalctl -xeu podman-auto-update`
- Or the same thing for user's rootless service instead: `journalctl --user -xeu podman-auto-update`
- **Important:** If the container fails to start after an image auto-update, then it will automatically be rolled back to the last working image version. This happens during `podman auto-update` itself, since that command is responsible for starting the server again. If it sees that the service cannot start, then it rolls back the changes. [See the documentation for more details](https://docs.podman.io/en/latest/markdown/podman-auto-update.1.html#rollback). You can ignore the documentation's notes about `--sdnotify`, since everything is already configured properly if you've used Podman Quadlets to set up the Systemd service.

PS: "daemon-reload" and other commands do not update the image. It doesn't even matter if you have a newer version of that image on disk already. The **only** thing that updates a container to use a newer version of an image is the `podman auto-update` task.

**Reply by dwedia** [reply]
Thank you for this reply. It helped me alot in finding my way around podman-auto-update :)
